django/contrib/admin/static/admin/js/SelectBox.js

```diff
@@ -8,23 +8,56 @@
             const cache = SelectBox.cache[id];
             const boxOptions = box.options;
             const boxOptionsLength = boxOptions.length;
-            for (let i = 0, j = boxOptionsLength; i < j; i++) {
-                const node = boxOptions[i];
-                cache.push({value: node.value, text: node.text, displayed: 1});
+            const noGroup = [];
+            if(box.children.length > 0) {
+                // optgroup not present
+                if(box.children[0].tagName.toLowerCase() === 'option') {
+                    for (let i = 0, j = boxOptionsLength; i < j; i++) {
+                        const node = boxOptions[i];
+                        noGroup.push({value: node.value, text: node.text, displayed: 1 });
+                    }
+                    cache.push({ 'group': null, 'items': noGroup });
+                }
+                // optgroup present
+                if(box.children[0].tagName.toLowerCase() === 'optgroup') {
+                    for (let p = 0; p < box.children.length; p++) { 
+                        const options = box.children[p].children;
+                        const group = [];
+                        for (let r = 0; r < options.length; r++) {
+                            group.push({ value: options[r].value, text: options[r].text, displayed: 1 });
+                        }
+                        cache.push({ 'group': box.children[p].label, 'items': group });
+                    }
+                }
             }
         },
         redisplay: function(id) {
             // Repopulate HTML select box from cache
             const box = document.getElementById(id);
             box.innerHTML = '';
             const cache = SelectBox.cache[id];
+            let referance;
             for (let i = 0, j = cache.length; i < j; i++) {
                 const node = cache[i];
-                if (node.displayed) {
-                    const new_option = new Option(node.text, node.value, false, false);
-                    // Shows a tooltip when hovering over the option
-                    new_option.title = node.text;
-                    box.appendChild(new_option);
+                if (node.items.length == 0) {
```

Comment:
Pleaes always use `===` in JavaScript. Please fix throughout the PR.

Reply:
replaced all `==` with `===` in SelectBox.js